src/pynwb/form/build/builders.py

```diff
@@ -184,7 +186,14 @@ def __set_builder(self, builder, obj_type):
         name = builder.name
         if name in self.obj_type:
             if self.obj_type[name] != obj_type:
-                raise KeyError("'%s' already exists as %s" % (name, self.obj_type[name]))
+                if name == 'comments':
+                    
+                    # "comments" exists as both an attribute and as dataset in some groups, in legacy files.  
+                    #     To allow read to get past this special case, this will skip the issue.
+                    warnings.warn("'%s' already exists as %s; skipping..." % (name, self.obj_type[name]))
+
+                else:
+                    raise KeyError("'%s' already exists as %s" % (name, self.obj_type[name]))
```

Comment:
This is a accomodation for legacy files, that is not isolated to the legacy module.  Apparently comments were allowed in multiple places in the original spec; this lets those get read.  A sub-optimal solution.